rotkehlchen/db/custom_assets.py

```diff
@@ -0,0 +1,111 @@
+from typing import TYPE_CHECKING, List, Tuple
+
+from rotkehlchen.assets.asset import CustomAsset
+from rotkehlchen.assets.types import AssetType
+from rotkehlchen.db.filtering import CustomAssetsFilterQuery
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.globaldb import GlobalDBHandler
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+class DBCustomAssets:
+    def __init__(self, db_handler: 'DBHandler') -> None:
+        self.db = db_handler
+
+    @staticmethod
+    def _get_custom_assets(filter_query: CustomAssetsFilterQuery) -> List[CustomAsset]:
+        """Queries the custom_assets table using the filter query and returns a list of `CustomAsset`."""  # noqa: E501
+        query, bindings = filter_query.prepare()
+        query = 'SELECT B.identifier, B.name, A.type, A.notes from custom_assets `A` JOIN assets `B` ON A.identifier = B.identifier ' + query  # noqa: E501
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            return [CustomAsset.deserialize_from_db(entry) for entry in cursor]
+
+    def get_custom_assets_and_limit_info(
+            self,
+            filter_query: CustomAssetsFilterQuery,
+    ) -> Tuple[List[CustomAsset], int, int]:
+        """
+        Returns a list of `CustomAsset`, a count of the assets that match
+        the query and the count of custom assets in the DB.
+        """
+        entries = self._get_custom_assets(filter_query=filter_query)
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            query, bindings = filter_query.prepare(with_pagination=False)
+            query = 'SELECT COUNT(*) from custom_assets AS A JOIN assets AS B ON A.identifier = B.identifier ' + query  # noqa: E501
+            entries_found = cursor.execute(query, bindings).fetchone()[0]
+            entries_total = cursor.execute('SELECT COUNT(*) FROM custom_assets;').fetchone()[0]
+            return entries, entries_found, entries_total
+
+    def add_custom_asset(self, custom_asset: CustomAsset) -> str:
+        """
+        Adds a custom new custom asset into `assets` and `custom_assets` table.
+        May raise:
+        - InputError if the combination of the name & type / asset with identifier already exist.
+        """
+        self.check_if_custom_asset_exists(custom_asset)
+        with GlobalDBHandler().conn.write_ctx() as global_db_write_cursor, self.db.user_write() as db_write_cursor:  # noqa: E501
```

Comment:
since the `db_write_cursor` is only use at the last function why not add it only there? I don't see why you need to open a context for user db for the 2 globalDB functions if you don't use the user db cursor in them.